### ISSUE TITLE
SQL CU updates

### ISSUE THREAD

**Issue by Chewwy420** [jimm98y/MSSQLEXPRESS-M1-Install#16]
Anyone finds a way to install the CU updates once SQL is installed?

**Comment by TroyT868** [comment]
I haven't had any issues installing them other than they report that they fail to upgrade the SQL Server engine when the installation wizard is done running but everything comes back up after a restart. I just got done installing SQL Server 2022 CU # 13 released on May 16th, 2024. I can't speak to SQL Server 2019 or any edition of SQL Server 2022 other than the Developer edition in Parallels Desktop which I have installed on my M3 MacBook Pro.

Basically I download the CU's setup file and save it to my Desktop. Then I run it as the administrator. When the pre-install checks run if the restart check fails then restart the VM from within Windows 11. Once that's done follow the wizard to install the CU. At the final screen it'll say that everything updated except for the SQL Server engine. When you look at the Services app you'll see that the SQL Server (MSSQLSERVER) service is in a stopped state but is still set to Automatic (Delayed Start) and if you try to start it you'll receive an error message but if you go ahead and restart the VM from within Windows 11 and allow it to restart then wait a for that service to start .... which doesn't happen right away because it's set to Automatic (Delayed Start) but in my case if I wait about 15 seconds then refresh the Services app or close it and re-open it will then show it's started .... I can then open SQL Server Management Studio, connect to the SQL instance as usual and sure enough the build number has updated to the CU I just installed.

I've done it with two different CU's now in that exact order right down to trying to manually start the service after the install only to get an error and by doing it in exactly in that order the CU gets installed and works despite what it reports at the end of the install wizard. I believe it fails in my case because the SQL Server Agent component doesn't install using the script on this website and I think the CU then the engine update fails because it can't update that component because it's not there in the first place. That's my best guess anyway, but even if I'm dead wrong and it shows it failed initially for some other reason, following the steps outlined above works for me.

On a side note I was running SQL Server Management Studio v19.3 prior to installing CU # 13, upgraded to the latest version of SQL Server Management Studio v20 and made sure that worked prior to installing CU #13 which I doubt made any kind of a difference because they are separate pieces of software that can exist without one another being installed, but in the interest of full disclose v20 was installed and working prior to upgrading to CU #13. I left v19.3 on the system as well. Again, I don't think it matters as I've never had an issue with installing updates to the v19 line or installing v20 of that software but that is how the machine was configured when I got CU #13 installed and working just a few minutes ago.
